### PR TITLE
[fields] Prevent keyboard editing when disabled

### PR DESCRIPTION
**Pull request by arthurbalduini**
Closes [#13387](https://github.com/mui/mui-x/issues/13387)

### CODE REVIEW

File: packages/x-date-pickers/src/internals/hooks/useField/useFieldV6TextField.ts [hunk 1]
```diff
@@ -273,7 +273,7 @@ export const useFieldV6TextField: UseFieldTextField<false> = (params) => {
     // prevent default to avoid the input `onChange` handler being called
     event.preventDefault();
 
-    if (readOnly) {
+    if (readOnly || disabled) {
```
Comment (LukasTy) [inline packages/x-date-pickers/src/internals/hooks/useField/useFieldV6TextField.ts:276]: Nice spot! 💙

File: packages/x-date-pickers/src/internals/hooks/useField/useFieldV6TextField.ts [hunk 2]
```diff
@@ -88,7 +88,7 @@ export const useFieldV6TextField: UseFieldTextField<false> = (params) => {
       inputRef: inputRefProp,
       placeholder: inPlaceholder,
     },
-    internalProps: { readOnly = false },
+    internalProps: { readOnly = false, disabled },
```
Comment (LukasTy) [inline packages/x-date-pickers/src/internals/hooks/useField/useFieldV6TextField.ts:91]: WDYT about aligning the approach and initializing the boolean to a default value as with `readOnly`?
```suggestion
    internalProps: { readOnly = false, disabled = false },
```

**Review by LukasTy — COMMENTED** [review]
Great job! 👏 
Thanks for taking care of it. 🙏 
Could you add at least a few tests for v7 fields (with `enableAccessibleFieldDOMStructure`) keyboard editing when disabled? 🙏 
Source: https://github.com/mui/mui-x/issues/13387#issuecomment-2150529057

File: packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx [hunk 3]
```diff
@@ -744,6 +744,65 @@ describe('<DateField /> - Editing', () => {
...
+      v7Response.selectSection('month');
+
+      keys.forEach((key) => {
+        v7Response.pressKey(0, key);
+        expectFieldValueV7(v7Response.getSectionsContainer(), 'MM/DD/YYYY');
+        expect(onChangeV7.callCount).to.equal(0);
+      });
+
+      // digit key press
+      // v7Response.pressKey(0, '2');
```
Comment (arthurbalduini) [inline packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx:780]: This event trigger still seems to be able to change the input value 🤯 
I'm investigating it, but accept suggestions 👼

File: packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx [hunk 4]
```diff
@@ -744,6 +744,63 @@ describe('<DateField /> - Editing', () => {
     });
   });
 
+  describeAdapters('Disabled field', DateField, ({ renderWithProps }) => {
```
Comment (arthurbalduini) [inline packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx:747]: Do you see a good reason to implement similar tests for `TimeField`, or should we test it elsewhere?

Comment (LukasTy) [reply packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx:747]: Good point, I agree, there is no need to duplicate the tests on other regular Field components. 👍

**Review by LukasTy — APPROVED** [review]
LGTM! 👍 
Great job! 💯 🚀

**Pull request merged**